server/repository/src/database_settings.rs

```diff
@@ -49,7 +49,20 @@ impl DatabaseSettings {
 #[cfg(all(not(feature = "postgres"), not(feature = "memory")))]
 impl DatabaseSettings {
     pub fn connection_string(&self) -> String {
-        self.database_name.clone()
+        let sqlite_suffix = |s: String| -> bool {
+            if s.len() < 8 {
+                return false;
+            }
+            let split = s.char_indices().nth_back(6).unwrap().0;
+            if &s[split..] == ".sqlite" {
+                return true;
+            }
+            return false;
+        };
+        match sqlite_suffix(self.database_name.clone()) {
+            true => self.database_name.clone(),
+            false => format!("{}.sqlite", self.database_name.clone()),
+        }
```

Comment:
Unfortunately this breaks existing installation, i.e. it doesn't use my existing `omsupply-database` file. I will add a comment to the issue...

Reply:
Can this be simplified to:
```rust
pub fn connection_string(&self) -> String {
        if self.database_name.ends_with(".sqlite") {
            self.database_name.clone()
        } else {
            format!("{}.sqlite", self.database_name)
        }
    }
```

Reply:
This is being done in sqlite feature though... so would the file extension be anything other than sqlite...?

Reply:
Yes if an existing installation uses the old database name...

Reply:
I'm assuming that the you're after the connection_string function which still calls to check if the db exists without the suffix like:

```
pub fn connection_string(&self) -> String {
        if self.database_name.ends_with(".sqlite") {
            // just use DB if name ends in .sqlite
            self.database_name.clone()
        } else {
            // first check if database exists on disk. If it does, we will use db filename as is without appending .sqlite
            let db_exists = self.check_db_exists();
            match db_exists {
                true => self.database_name.clone(),
                false => format!("{}.sqlite", self.database_name.clone()),
            }
        }
    }
    ```